http/index.php

```diff
@@ -350,7 +350,10 @@ function writeMatrix(array $glVersions, SimpleXMLElement $xml, Mesamatrix\Hints
                         </thead>
                         <tbody>
 <?php
-foreach ($xml->commits->commit as $commit) {
+$numCommits = \Mesamatrix::$config->getValue('info', 'commitlog_length', 10);
+$numCommits = min($numCommits, $xml->commits->commit->count());
+for ($i = 0; $i < $numCommits; ++$i) {
```

Comment:
Use `array_slice()` here with offset `0` and length `$numCommits` to avoid the nasty raw loop, then you can use the `foreach` again.